server/db/migrations/patch-1-2.sql

```diff
@@ -0,0 +1,66 @@
+-- create the users table
+CREATE TABLE IF NOT EXISTS users (
+  user_id CHAR(32) PRIMARY KEY,
+  email VARCHAR(255) NOT NULL,
+  oauth_token TEXT,
+  created_at TIMESTAMPTZ(3) NOT NULL,
+  updated_at TIMESTAMPTZ(3) NOT NULL
+);
+-- create the user_pages table
+CREATE TABLE IF NOT EXISTS user_pages (
+  id UUID PRIMARY KEY,
+  user_id CHAR(32) REFERENCES users(user_id),
+  url VARCHAR(2048) NOT NULL,
+  raw_url VARCHAR(2048) NOT NULL,
+  url_hash CHAR(40) NOT NULL,
+  title TEXT NOT NULL,
+  extracted_at TIMESTAMPTZ(3),
+  extracted_author_name TEXT,
+  extracted_author_url VARCHAR(2048),
+  extracted_cache_age BIGINT,
+  extracted_content TEXT,
+  extracted_description TEXT,
+  extracted_favicon_color TEXT,
+  extracted_favicon_url VARCHAR(2048),
+  extracted_image_caption TEXT,
+  extracted_image_color TEXT,
+  extracted_image_entropy DOUBLE PRECISION,
+  extracted_image_height INTEGER,
+  extracted_image_url VARCHAR(2048),
+  extracted_image_width INTEGER,
+  extracted_language TEXT,
+  extracted_lead TEXT,
+  extracted_media_duration INTEGER,
+  extracted_media_height INTEGER,
+  extracted_media_html TEXT,
+  extracted_media_type TEXT,
+  extracted_media_width INTEGER,
+  extracted_provider_display TEXT,
+  extracted_provider_name TEXT,
+  extracted_provider_url VARCHAR(2048),
+  extracted_published TIMESTAMPTZ(3),
+  extracted_safe BOOLEAN,
```

Comment:
Unrelated to life, but me thinks we may need another couple columns for when `extracted_safe` is `false`:

- `extracted_safe_type` &mdash; either "phishing" or "malware".
- `extracted_safe_message` &mdash; a user-readable message in HTML format explaining that the given url is likely malicious.

(per http://embed.ly/docs/api/extract/features/safe)